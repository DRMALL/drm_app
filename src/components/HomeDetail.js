import React, { Component } from 'react'
import { View, Text, Image, ScrollView, TouchableOpacity, StatusBar } from 'react-native'
import Swiper from 'react-native-swiper'
import moment from 'moment'
import { primaryColor, mainColor, loginBorderColor, loginBackgroundColor } from '../common/constants'
import {  } from '../common/strings'
import { seekDetail, detail } from '../styles'
import { checkToken } from '../utils/handleToken'
import { getPort } from '../utils/fetchMethod'
import { getNewsOne } from '../apis'

const gobackWhiteIcon = require('../images/navigation_icons/goback_white.png')
const shareIcon = require('../images/navigation_icons/share.png')
const icInsertPhotoIcon = require('../images/navigation_icons/ic_insert_photo.png')
const uploadPic = require('../images/uploadPic.png')

export default class HomeDetail extends Component {
  static navigationOptions = {
    headerStyle: {
      height: 0,
      top: 50,
    }
  };

  constructor(props) {
    super(props)
    this.state = {
      newsOneData: {},
    }
  }

  componentDidMount() {
    this.getNewsOne()
  }

  getNewsOne() {
    let id = this.props.navigation.state.params.newsId
    checkToken('drmAppToken')
    .then(async token => {
      let res = await getPort(`${getNewsOne}?id=${id}&token=${token}`)
      if(res.code == 201) {
        this.setState({
          newsOneData: res.data,
        })
      }
    })
  }

  render() {
    let { navigation } = this.props
      , { newsOneData } = this.state
    return (
      <ScrollView style={{backgroundColor: mainColor}}>
        <StatusBar hidden={true}/>
        <HomeSwiperHeader picData={newsOneData.images ? newsOneData.images : []} navigation={navigation} />
        <Text style={[detail.titleText, {paddingHorizontal: 16}]}>{newsOneData.abstract}</Text>
        <Text style={[detail.titleTime, {paddingHorizontal: 16, paddingTop: 15}]}>
          {
            newsOneData.publish_time ? moment(newsOneData.publish_time).format('YYYY-MM-DD') : '0000-00-00'
          }
        </Text>
        <Text style={detail.ordinaryText}>
          {newsOneData.content}
        </Text>
      </ScrollView>
    )
  }
}

const HomeSwiperHeader = props => {
  let { picData, navigation } = props
  return (
    <View style={seekDetail.headerView}>
      <Swiper height={230} horizontal={true} dotColor={loginBorderColor} activeDotColor={mainColor}>
        {renderSwiper(picData)}
      </Swiper>
      <TouchableOpacity style={seekDetail.gobackIcon} onPress={() => navigation.goBack()}>
        <Image source={gobackWhiteIcon}/>
      </TouchableOpacity>
      <TouchableOpacity style={seekDetail.shareIcon} onPress={() => alert('share')}>
        <Image source={shareIcon}/>
      </TouchableOpacity>
    </View>
  )
}

const renderSwiper = (picData)=> {
  return (
    // picData[0] ? 
    picData.map((picItem, index)=> <View key={index} style={seekDetail.picsView}>
      <Image style={seekDetail.pics} source={{uri: picItem.url}}/>
    </View>)
     // : <TouchUploadPic />
  )
}

const TouchUploadPic =  props => {
  let { navigation } = props
  return (
    <TouchableOpacity style={detail.picsView} activeOpacity={0.8} disabled={true} onPress={()=> alert('upload')}>
      <Image style={[detail.pics, {resizeMode: 'center', backgroundColor: loginBackgroundColor}]} source={icInsertPhotoIcon}/>
    </TouchableOpacity>
  )
}


// {
//           picData[0] ? <View style={seekDetail.picsView}>
//             <Image style={seekDetail.pics} source={{uri: picData[0].url}}/>
//           </View> : <TouchUploadPic />
//         }
//         {
//           picData[1] ? <View style={seekDetail.picsView}>
//             <Image style={seekDetail.pics} source={{uri: picData[1].url}}/>
//           </View> : <TouchUploadPic />
//         }
//         {
//           picData[2] ? <View style={seekDetail.picsView}>
//             <Image style={seekDetail.pics} source={{uri: picData[2].url}}/>
//           </View> : <TouchUploadPic />
//         }
//         {
//           picData[3] ? <View style={seekDetail.picsView}>
//             <Image style={seekDetail.pics} source={{uri: picData[3].url}}/>
//           </View> : <TouchUploadPic />
//         }
//         {
//           picData[4] ? <View style={seekDetail.picsView}>
//             <Image style={seekDetail.pics} source={{uri: picData[4].url}}/>
//           </View> : <TouchUploadPic />
//         }