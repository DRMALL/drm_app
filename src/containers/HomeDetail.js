import React, { Component } from 'react'
import { View, Text, Image, ScrollView, TouchableOpacity, StatusBar, Platform } from 'react-native'
import moment from 'moment'
import { primaryColor, mainColor, loginBorderColor, loginBackgroundColor } from '../common/constants'
import {  } from '../common/strings'
import Loading2 from '../components/units/Loading2'
import ShareModal from '../components/units/ShareModal'
import HomeDetailWebView from '../components/home/HomeDetailWebView'
import { detail } from '../styles'

import HomeSwiperHeader from '../components/home/HomeSwiperHeader'
import store from '../utils/store'
import homeDetailAC from '../actions/homeDetailAC'
import getNewsOne from '../funcs/home/getNewsOne'

const gobackWhiteIcon = require('../images/navigation_icons/goback_white.png')
const shareIcon = require('../images/navigation_icons/share.png')
const icInsertPhotoIcon = require('../images/navigation_icons/ic_insert_photo.png')
const uploadPic = require('../images/uploadPic.png')

export default class HomeDetail extends Component {
  static navigationOptions = {
    headerStyle: {
      height: 0,
      top: 50,
    }
  };

  constructor(props) {
    super(props)
    this.state = store.getState().home
  }

  componentDidMount() {
    homeDetailAC.getOneData({})
    getNewsOne(this.props)
  }

  componentWillMount() {
    this.unsubscribe = store.subscribe( ()=> this.setState(store.getState().home) )
  }

  componentWillUnmount(){
    this.unsubscribe()
  }

  render() {
    let { navigation } = this.props
      , { newsOneData, shareShow, topView, nextView, height } = this.state
      , { hiddenShare } = homeDetailAC
      , contentLength = newsOneData.content ? newsOneData.content.split('').length : 0
    return (
      <View style={Platform.OS === 'ios' ? {top: -20, height: '103.5%'} : {height: '100%'}}>
        <ScrollView style={[{backgroundColor: mainColor}, shareShow ? nextView : topView]}>
          <StatusBar hidden={true}/>
          {
            !newsOneData.images ? <Loading2 animating={!newsOneData.images ? true : false} /> : 
            <HomeSwiperHeader picData={newsOneData.images} navigation={navigation} />
          }
          <Text style={[detail.titleText, {paddingHorizontal: 16}]}>{newsOneData.abstract}</Text>
          <Text style={[detail.titleTime, {paddingHorizontal: 16, paddingTop: 15}]}>
            {
              newsOneData.publish_time ? moment(newsOneData.publish_time).format('YYYY-MM-DD') : '0000-00-00'
            }
          </Text>
          <HomeDetailWebView 
            contentLength={contentLength} 
            content={newsOneData.content} 
          />
        </ScrollView>
        <ShareModal state={this.state} pressShareCancel={hiddenShare}/>
      </View>
    )
  }
}

// <Text style={detail.ordinaryText}>
//             {newsOneData.content}
//           </Text>
// const TouchUploadPic =  props => {
//   let { navigation } = props
//   return (
//     <TouchableOpacity style={detail.picsView} activeOpacity={0.8} disabled={true} onPress={()=('upload')}>
//       <Image style={[detail.pics, {resizeMode: 'center', backgroundColor: loginBackgroundColor}]} source={icInsertPhotoIcon}/>
//     </TouchableOpacity>
//   )
// }
