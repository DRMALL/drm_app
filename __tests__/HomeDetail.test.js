import 'react-native'
import React from 'react'
import HomeDetail from '../src/containers/HomeDetail'
import renderer from 'react-test-renderer';

jest.unmock('ScrollView')    //RN WebView(TypeError: Cannot read property 'decelerationRate' of undefined)

test('HomeDetail test', () => {
  const navigation = {
    state: {
      params: {
        newsId: 'id',
      },
    },
  }
  const tree = renderer.create(
    <HomeDetail navigation={ navigation } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})