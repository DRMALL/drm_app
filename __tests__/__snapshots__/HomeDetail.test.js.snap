// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeDetail test 1`] = `
<View
  style={
    Object {
      "height": "103.5%",
      "top": -20,
    }
  }
>
  <RCTScrollView
    alwaysBounceHorizontal={undefined}
    alwaysBounceVertical={true}
    onContentSizeChange={null}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={undefined}
    onResponderTerminationRequest={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollShouldSetResponder={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    scrollEventThrottle={undefined}
    sendMomentumEvents={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "overflow": "scroll",
        },
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "position": "relative",
            "zIndex": 3,
          },
        ],
      ]
    }
  >
    <RCTScrollContentView
      collapsable={false}
      removeClippedSubviews={undefined}
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 220,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "height": 50,
            }
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(0,0,0,.8)",
              "fontSize": 20,
            },
            Object {
              "paddingHorizontal": 16,
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 12,
            },
            Object {
              "paddingHorizontal": 16,
              "paddingTop": 15,
            },
          ]
        }
      >
        0000-00-00
      </Text>
      <View
        style={
          Object {
            "height": 0,
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTWebView
            allowsInlineMediaPlayback={undefined}
            automaticallyAdjustContentInsets={false}
            bounces={undefined}
            contentInset={undefined}
            dataDetectorTypes={undefined}
            decelerationRate={0.99}
            injectedJavaScript="
              (function () {
                const arr = document.getElementsByTagName(\\"img\\")
                let height = null
                for (let i = 0; i < arr.length; i ++) {
                  arr[i].width = 690
                }
                function changeHeight() {
                  if (document.body.scrollHeight != height) {
                    height = document.body.scrollHeight
                    if (window.postMessage) {
                      window.postMessage(JSON.stringify({
                        type: 'setHeight',
                        height: height,
                      }))
                    }
                  }
                }
                setInterval(changeHeight, 100)
              } () )"
            mediaPlaybackRequiresUserAction={undefined}
            messagingEnabled={true}
            onLoadingError={[Function]}
            onLoadingFinish={[Function]}
            onLoadingStart={[Function]}
            onMessage={[Function]}
            onShouldStartLoadWithRequest={undefined}
            scalesPageToFit={undefined}
            scrollEnabled={false}
            source={
              Object {
                "html": undefined,
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
                Object {
                  "height": "100%",
                },
              ]
            }
          />
        </View>
      </View>
    </RCTScrollContentView>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "position": "relative",
          "zIndex": 3,
        },
        Object {
          "display": "none",
          "height": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "black",
          "height": "100%",
          "opacity": 0.5,
          "position": "relative",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#F7F7F7",
          "bottom": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "paddingHorizontal": 16,
            "paddingVertical": 10,
          }
        }
      >
        分享到
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginRight": 20,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={1}
            style={
              Object {
                "height": 66,
                "resizeMode": "contain",
                "width": 66,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 14,
                "paddingVertical": 10,
              }
            }
          >
            微信
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginRight": 20,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={1}
            style={
              Object {
                "height": 66,
                "resizeMode": "contain",
                "width": 66,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 14,
                "paddingVertical": 10,
              }
            }
          >
            微博
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "rgba(0,0,0,.06)",
            "borderWidth": 0.5,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "paddingVertical": 10,
            }
          }
        >
          取消
        </Text>
      </View>
    </View>
  </View>
</View>
`;
